src/messaging/ExchangeContext.cpp

```diff
@@ -88,6 +88,7 @@ void ExchangeContext::SetResponseTimeout(Timeout timeout)
 void ExchangeContext::UpdateSEDPollingMode()
 {
     Transport::PeerAddress address;
+    VerifyOrReturn(mSession);
```

Comment:
But actually... if we have no session handle, shouldn't we in fact call `UpdateSEDPollingMode(false)`?  Otherwise it seems like we'll be stuck in polling fast-poll mode?

Reply:
We have a safety mechanism in `~ExchangeContext()` to always release the fast-polling mode request.

Reply:
Ok, I believed that MRP would still continue to work even after the session deletion, but it appears it would not, so there's not point in keeping the fast-polling mode. I added code to release it here.